internal/business/disable_method/disable_method.go

```diff
@@ -0,0 +1,43 @@
+package disable_method
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"net/http"
+)
+
+var methodCounter = prometheus.NewCounterVec(prometheus.CounterOpts{
+	Namespace: "go_graphql_armor",
+	Subsystem: "disable_get_method",
+	Name:      "count",
+	Help:      "Amount of times the disable method rule was triggered",
+},
+	[]string{},
+)
+
+func init() {
+	prometheus.MustRegister(methodCounter)
+}
+
+type Config struct {
+	Enabled bool `conf:"default:true" yaml:"enabled"`
+}
+
+func DisableMethodRule(cfg Config) func(next http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		fn := func(w http.ResponseWriter, r *http.Request) {
+			if !cfg.Enabled {
+				next.ServeHTTP(w, r)
+				return
+			}
+
+			if r.Method == "GET" {
```

Comment:
a case could also be made for explicitly allowing POST and disallowing the rest?

Reply:
That's fair. I based this off of https://graphql.org/learn/serving-over-http/#http-methods-headers-and-body which states either GET or POST is supported.

The rule is now disable GET, assuming POST is the only one left. If I were to adopt the POST-only strategy the rule should probably be called differently

Reply:
I see the WIP official HTTP specification [states that POST](https://github.com/graphql/graphql-over-http/blob/main/spec/GraphQLOverHTTP.md#request) must be supported and other methods _may_ be supported. I like your suggestion and will adopt it in a change, mostly considering clear naming for the feature at the moment.